src/app/routes/applications/new-role/services/role-creation.service.ts

```diff
@@ -21,10 +21,15 @@ export class RoleCreationService {
   async canUseDomain(domain: string): Promise<boolean> {
     try {
       const exists = await this.domainsFacade.checkExistenceOfDomain(domain);
-      if (!exists) {
-        return true;
+      const isOwner = await this.domainsFacade.isOwner(domain);
+      if (exists) {
+        return false;
+      }
+      if (!exists && !isOwner) {
+        return false;
       }
-      return await this.domainsFacade.isOwner(domain);
+
+      return !exists;
```

Comment:
I'm confused, it seems to me like the effective implementation stayed the same 🤔 
Isn't the new logic effectively the same as the old logic?

Reply:
nope, i updated the tests and they were failing. Previous set of tests and implementation was checking if domain exist, here method was saying canUseDomain.
So used functions are the same, but the logic and returned values are different.

Reply:
I don't understand how that's possible 🤔 .
My assumption is that `!exists` will always be `true` here because of this clause 
```
if (exists) {
  return false;
}
```
So we can substitute `!exists` for `true`. Then we have
```
if (true && !isOwner) {
        return false;
}
return true;
```
Then, by removing the redundant `true` and inverting the `if` statement we get
```
if (isOwner) {
 return true;
}
return false;
``` 
This can be simplified to `return isOwner`. If you remove the assignment to `isOwner`, then we have line currently on `develop`. What am I missing here?

Reply:
Yeah, you're right. I removed most of the code, and tests are passing.